SU2_CFD/include/variables/CTurbSSTVariable.hpp

```diff
@@ -46,58 +46,54 @@
  * \author A. Bueno.
  */
 
-class CTurbSSTVariable : public CTurbVariable {
+class CTurbSSTVariable final : public CTurbVariable {
 protected:
-  su2double sigma_om2,
-  beta_star;
-  su2double F1,    /*!< \brief Menter blending function for blending of k-w and k-eps. */
-  F2,            /*!< \brief Menter blending function for stress limiter. */
-  CDkw;           /*!< \brief Cross-diffusion. */
+  su2double sigma_om2;
+  su2double beta_star;
+  Vec_t F1;
+  Vec_t F2;    /*!< \brief Menter blending function for blending of k-w and k-eps. */
+  Vec_t CDkw;  /*!< \brief Cross-diffusion. */
 
 public:
   /*!
    * \brief Constructor of the class.
-   */
-  CTurbSSTVariable(void);
-
-  /*!
-   * \overload
-   * \param[in] val_rho_kine - Turbulent variable value (initialization value).
-   * \param[in] val_rho_omega - Turbulent variable value (initialization value).
-   * \param[in] val_muT - Turbulent variable value (initialization value).
-   * \param[in] val_nDim - Number of dimensions of the problem.
-   * \param[in] val_nvar - Number of variables of the problem.
+   * \param[in] kine - Turbulence kinetic energy (k) (initialization value).
+   * \param[in] omega - Turbulent variable value (initialization value).
+   * \param[in] mut - Eddy viscosity (initialization value).
+   * \param[in] npoint - Number of points/nodes/vertices in the domain.
+   * \param[in] ndim - Number of dimensions of the problem.
+   * \param[in] nvar - Number of variables of the problem.
    * \param[in] constants -
    * \param[in] config - Definition of the particular problem.
    */
-  CTurbSSTVariable(su2double val_rho_kine, su2double val_rho_omega, su2double val_muT, unsigned short val_nDim, unsigned short val_nvar,
-                   su2double *constants, CConfig *config);
-
+  CTurbSSTVariable(su2double kine, su2double omega, su2double mut, Idx_t npoint,
+                   Idx_t ndim, Idx_t nvar, const su2double* constants, CConfig *config);
+                   
   /*!
    * \brief Destructor of the class.
    */
-  ~CTurbSSTVariable(void);
+  ~CTurbSSTVariable() = default;
```

Comment:
Are all of these C++11 keywords necessary for functioning and/or performance ?

Reply:
`final` might be as the compiler will then know which virtual methods will no longer be redefined.
`override` was to guarantee that I was indeed overriding the virtual methods of base (easy mistake to make when changing function signatures). `default` means {} in that case. I think those keywords are good documentation for someone reading the code.

Reply:
Thanks for the explanation! We should use these keywords whenever we can.